### ISSUE TITLE
Question about loss_distill

### ISSUE THREAD

**Issue by Sundrops** [clovaai/overhaul-distillation#10]
The distillation_loss between last s_feats and last t_feats will be 1, as shown in https://github.com/clovaai/overhaul-distillation/blob/master/ImageNet/distiller.py#L69.
Why distillation_loss **\*\*(feat_num - i - 1)**? Is aligning the Layer 1 is more important than Layer 4?
```py
for i in range(feat_num):
    s_feats[i] = self.Connectors[i](s_feats[i])
    loss_distill += distillation_loss(s_feats[i], t_feats[i].detach(), getattr(self, 'margin%d' % (i + 1))) \
                    / 2 ** (feat_num - i - 1)
```

**Comment by bhheo** [comment]
Hi, thank you for your interest in my study.

The power operator ( ** ) has priority than division operator ( / ) 
For example,
``` Python
3 / 2 ** 3
>> 0.375
```
So, it is `distillation loss / (2^(feat_num - i  - 1))`

Detail about this parameter was explained in #8

**Comment by Sundrops** [comment]
Got it. Thanks for your reply.

**Comment by Sundrops** [comment]
Maybe it will be better to directly cal mean of loss instead of sum and adjust the value of alpha accordingly. Thanks  for your  great work.
```py
def distillation_loss(source, target, margin):
    loss = ((source - margin)**2 * ((source > margin) & (target <= margin)).float() +
            (source - target)**2 * ((source > target) & (target > margin) & (target <= 0)).float() +
            (source - target)**2 * (target > 0).float())
    return torch.abs(loss).sum() # -> torch.abs(loss).mean()
```

**Comment by bhheo** [comment]
Yes. I agree with you.
I realized that after I adjusted alpha for sum operation.

But, in the case of mean, loss per channel is smaller than sum when the base channel of network is large.
So I recommend to multiply the base channel size to mean and adjust alpha.
```
torch.abs(loss).mean() * base_channel
```

**Comment by Sundrops** [comment]
Yes. Thanks for your careful explanation!
